pkg/query-service/app/logs/v4/query_builder.go

```diff
@@ -29,3 +35,492 @@ const (
 	DISTRIBUTED_LOGS_V2_RESOURCE = "distributed_logs_v2_resource"
 	NANOSECOND                   = 1000000000
 )
+
+func getClickhouseLogsColumnDataType(columnDataType v3.AttributeKeyDataType) string {
+	if columnDataType == v3.AttributeKeyDataTypeFloat64 || columnDataType == v3.AttributeKeyDataTypeInt64 {
+		return "number"
+	}
+	if columnDataType == v3.AttributeKeyDataTypeBool {
+		return "bool"
+	}
+	return "string"
+}
+
+func getClickhouseKey(key v3.AttributeKey) string {
+	// check if it is a top level static field
+	if _, ok := constants.StaticFieldsLogsV3[key.Key]; ok && key.Type == v3.AttributeKeyTypeUnspecified {
+		return key.Key
+	}
+
+	//if the key is present in the topLevelColumn then it will be only searched in those columns,
+	//regardless if it is indexed/present again in resource or column attribute
+	if !key.IsColumn {
+		columnType := logsV3.GetClickhouseLogsColumnType(key.Type)
+		columnDataType := getClickhouseLogsColumnDataType(key.DataType)
+		return fmt.Sprintf("%s_%s['%s']", columnType, columnDataType, key.Key)
+	}
+
+	// materialized column created from query
+	return "`" + utils.GetClickhouseColumnNameV2(string(key.Type), string(key.DataType), key.Key) + "`"
```

Comment:
materialized columns won't have any dots in them. Why do you need "``" around the column name?

Reply:
So this is the reason why we added '`' to all the materialized columns while referencing them. Let me add it as a comment.

Reply:
https://github.com/SigNoz/signoz/pull/4775